src/js/fable-core/Char.ts

```diff
@@ -116,90 +117,84 @@ const isSymbolMask = 0
   | 1 << UnicodeCategory.CurrencySymbol
   | 1 << UnicodeCategory.ModifierSymbol
   | 1 << UnicodeCategory.OtherSymbol;
-const isWhiteSpaceMask = 0
-  | 1 << UnicodeCategory.SpaceSeparator
-  | 1 << UnicodeCategory.LineSeparator
-  | 1 << UnicodeCategory.ParagraphSeparator;
 
 export const getUnicodeCategory = getCategory();
 
-export function isControl(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
+export function isControl(s: string, index?: number) {
+  const test = 1 << getUnicodeCategory(s, index);
   return (test & isControlMask) !== 0;
 }
 
-export function isDigit(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
+export function isDigit(s: string, index?: number) {
+  const test = 1 << getUnicodeCategory(s, index);
   return (test & isDigitMask) !== 0;
 }
 
-export function isLetter(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
+export function isLetter(s: string, index?: number) {
+  const test = 1 << getUnicodeCategory(s, index);
   return (test & isLetterMask) !== 0;
 }
 
-export function isLetterOrDigit(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
+export function isLetterOrDigit(s: string, index?: number) {
+  const test = 1 << getUnicodeCategory(s, index);
   return (test & isLetterOrDigitMask) !== 0;
 }
 
-export function isUpper(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
+export function isUpper(s: string, index?: number) {
+  const test = 1 << getUnicodeCategory(s, index);
   return (test & isUpperMask) !== 0;
 }
 
-export function isLower(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
+export function isLower(s: string, index?: number) {
+  const test = 1 << getUnicodeCategory(s, index);
   return (test & isLowerMask) !== 0;
 }
 
-export function isNumber(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
+export function isNumber(s: string, index?: number) {
+  const test = 1 << getUnicodeCategory(s, index);
   return (test & isNumberMask) !== 0;
 }
 
-export function isPunctuation(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
+export function isPunctuation(s: string, index?: number) {
+  const test = 1 << getUnicodeCategory(s, index);
   return (test & isPunctuationMask) !== 0;
 }
 
-export function isSeparator(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
+export function isSeparator(s: string, index?: number) {
+  const test = 1 << getUnicodeCategory(s, index);
   return (test & isSeparatorMask) !== 0;
 }
 
-export function isSymbol(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
+export function isSymbol(s: string, index?: number) {
+  const test = 1 << getUnicodeCategory(s, index);
   return (test & isSymbolMask) !== 0;
 }
 
-export function isWhiteSpace(cp: number) {
-  const test = 1 << getUnicodeCategory(cp);
-  return ((test & isWhiteSpaceMask) !== 0)
-    || (0x09 <= cp && cp <= 0x0D) || cp === 0x85 || cp === 0xA0;
+export function isWhiteSpace(s: string, index?: number) {
```

Comment:
@alfonsogarciacaro Shouldn't `isWhiteSpace` be using `getUnicodeCategory(s, index)` instead of regex?